### PR TITLE
Wires

### PR DESCRIPTION
**Pull request by afdw**

### CODE REVIEW

File: common/buildcraft/transport/item/ItemWire.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import java.util.List;
+
+public class ItemWire extends ItemBC_Neptune {
+    public ItemWire(String id) {
+        super(id);
```
Comment (AlexIIL) [inline common/buildcraft/transport/item/ItemWire.java]: If you add ````setHasSubtypes(true)```` then different types of wires won't stack

Comment (afdw) [inline common/buildcraft/transport/item/ItemWire.java]: Currently different types of wires doesn't stack, but OK, I'll add

Comment (AlexIIL) [inline common/buildcraft/transport/item/ItemWire.java]: Oh I mean when i picked them up off the ground - sorry.

**Comment by afdw** [comment]
@AlexIIL I think now it's ready to check how wires are working

**Comment by AlexIIL** [comment]
@afdw Ok, I'll clone it and start testing. :)

**Comment by afdw** [comment]
@AlexIIL OK, I tested performance.
1. Old system (5x5x5, 1 wire per block): 
2. New system (10x10x10, 8 wires per block): 
3. New system (5x5x5, 8 wires per block):

File: common/buildcraft/transport/statements/TriggerPipeSignal.java [hunk 2]
```diff
@@ -66,12 +67,14 @@ public boolean isTriggerActive(IStatementContainer container, IStatementParamete
...
         }
 
         for (IStatementParameter param : parameters) {
             if (param != null && param instanceof TriggerParameterSignal) {
                 TriggerParameterSignal signal = (TriggerParameterSignal) param;
                 if (signal.colour != null) {
+                    return wires.isAnyPowered(colour);
                     // FIXME: replace these calls with the correct ones in IWireManager!
                     // if (!wireManager.isWireOn(signal.colour)) {
                     // return false;
```
Comment (AlexIIL) [inline common/buildcraft/transport/statements/TriggerPipeSignal.java]: You should replace every ````wires.isAnyPowered(colour)```` with ````wireManager.isWireOn(signal.colour)```` in the above commented out logic.

Comment (AlexIIL) [inline common/buildcraft/transport/statements/TriggerPipeSignal.java]: woops, I swapped the order around :/

**Comment by afdw** [comment]
@AlexIIL Can you try to reproduce bug with removing wires now?

**Pull request merged**